src/test/java/com/rtbhouse/kafka/workers/integration/ProcessingFailureTest.java

```diff
@@ -91,7 +91,9 @@ public void onShutdown(WorkersException exception) {
         // then
         assertThat(latch.await(10, TimeUnit.SECONDS)).isTrue();
         assertThat(exceptions.size()).isEqualTo(1);
-        assertThat(exceptions.get(0)).isExactlyInstanceOf(ProcessingFailureException.class);
+        Throwable t;
+        for (t = exceptions.get(0); t.getCause() != null; t = t.getCause());
+        assertThat(t).isExactlyInstanceOf(ProcessingFailureException.class);
```

Comment:
here we should throw our own exception and check if our exception have been thrown